sources/packages/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -0,0 +1,237 @@
+   * to determine the associated program.
+   * @param offeringId offering id, when available, otherwise
+   * a PIR request need happen to an offering id be provided.
+   * @returns program info update result.
+   */
+  async updateProgramInfo(
+    assessmentId: number,
+    locationId: number,
+    status: ProgramInfoStatus,
+    programId?: number,
+    offeringId?: number,
+  ): Promise<StudentAssessment> {
+    const assessment = await this.repo
+      .createQueryBuilder("assessment")
+      .select(["assessment.id", "application.id"])
+      .innerJoin("assessment.application", "application")
+      .where("assessment.id = :assessmentId", { assessmentId })
+      .getOne();
+
+    if (!assessment) {
+      throw new CustomNamedError(
+        `Not able to find the assessment id ${assessmentId}`,
+        ASSESSMENT_NOT_FOUND,
+      );
+    }
+
+    assessment.application.location = { id: locationId } as InstitutionLocation;
+    assessment.application.pirProgram = { id: programId } as EducationProgram;
+    assessment.application.pirStatus = status;
+    assessment.offering = {
+      id: offeringId,
+    } as EducationProgramOffering;
+    // TODO: Kept for backward compatibility. It will be removed in an upcoming PR.
+    assessment.application.offering = {
+      id: offeringId,
+    } as EducationProgramOffering;
+
+    return this.repo.save(assessment);
+  }
+
+  /**
+   * Updates the assessment workflowId.
+   * The workflow id on the assessment table must be null otherwise
+   * the update will not be performed. Avoiding the update prevents
+   * that two different workflow instances try to process the same
+   * assessment.
+   * @param assessmentId assessment id to be updated.
+   * @param assessmentWorkflowId workflowId to be updated.
+   */
+  async updateWorkflowId(
+    assessmentId: number,
+    assessmentWorkflowId: string,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: assessmentId, assessmentWorkflowId: IsNull() },
+      { assessmentWorkflowId },
+    );
+  }
+
+  /**
+   * Starts the assessment workflow of one Student Application.
+   * @param assessmentId Student assessment that need to be processed.
+   */
+  async startAssessment(assessmentId: number): Promise<void> {
+    const assessment = await this.repo
+      .createQueryBuilder("assessment")
+      .select([
+        "assessment.id",
+        "assessment.assessmentWorkflowId",
+        "assessment.triggerType",
+        "application.id",
+        "application.applicationStatus",
+        "application.data",
+      ])
+      .innerJoin("assessment.application", "application")
+      .where("assessment.id = :assessmentId", { assessmentId })
+      .getOne();
+
+    if (assessment.assessmentWorkflowId) {
+      throw new CustomNamedError(
+        `Student assessment was already started and has a workflow associated with. Assessment id ${assessmentId}`,
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
```

Comment:
The custom error is appropriate to the actual error defined, but it would be better to have a specific named error like "ASSESSMENT_ALREADY_ASSOCIATED" or "ASSESSMENT_FOR_THE_APPLICATION_ASSOCIATED"

Reply:
We can be more specific, yes, but there will be no difference for the consumer of the constants because they all say the same: "It was not possible to execute the process". If we need a specific reason to differentiate them I would besupporting the creation of the specific ones. Please, let me know your thoughts.
Actually, the _INVALID_OPERATION_IN_THE_CURRENT_STATUS_ was supposed to be _INVALID_OPERATION_IN_THE_CURRENT_STATE_.